src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -141,10 +153,8 @@ private ImmutableArray<ITypeSymbol> GetTypeSymbols(ImmutableArray<ISymbol> candi
                             continue;
                         }
 
-                        type = parameters[ordinalInLambda].Type;
+                        builder.Add(parameters[ordinalInLambda].Type);
```

Comment:
When this was outside the `if (type.IsDelegateType())`, we would mix in regular non-delegate parameter types as though they were a type argument to a delegate type. So `Action<int>` would unwrap to `int`, and `string` wouldn't unwrap, but would still be included in the results.